scene/resources/capsule_shape_2d.cpp

```diff
@@ -60,6 +60,9 @@ void CapsuleShape2D::_update_shape() {
 
 void CapsuleShape2D::set_radius(real_t p_radius) {
 	radius = p_radius;
+	if (radius > height * 0.5) {
+		height = radius * 2;
```

Comment:
When working on the 3D version, I noticed that modifying the height while changing the radius can cause discrepancies in the editor undo/redo.

What about just clamping with `radius = height * 0.5;` in this case? I can fix that along with the 3D version, or you can make a separate PR if you prefer.

Reply:
#51527